components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -1297,6 +1308,179 @@ public void changeApplicationOwner(String applicationId, ApplicationOwner applic
+            if (apiResource == null) {
+                throw buildClientError(ErrorMessage.API_RESOURCE_NOT_FOUND, apiId, tenantDomain);
+            }
+            validateAPIResourceScopes(apiResource, addedScopes);
+
+            // Remove already authorized scopes from the added scopes list.
+            AuthorizedAPI currentAuthorizedAPI = getAuthorizedAPIManagementService().getAuthorizedAPI(applicationId,
+                    apiId, tenantDomain);
+            if (currentAuthorizedAPI == null) {
+                throw buildClientError(ErrorMessage.AUTHORIZED_API_NOT_FOUND, apiId, applicationId);
+            }
+            addedScopes.removeIf(scopeName -> currentAuthorizedAPI.getScopes().stream().anyMatch(scope ->
+                    scope.getName().equals(scopeName)));
+
+            getAuthorizedAPIManagementService().patchAuthorizedAPIs(applicationId, apiId, addedScopes, removedScopes,
+                    tenantDomain);
+
+        } catch (APIResourceMgtException e) {
+            String msg = "Error while fetching API resource with id: " + apiId;
+            throw Utils.buildServerError(msg, e);
+        } catch (IdentityApplicationManagementException e) {
+            String msg = "Error while updating authorized API with id: " + apiId + " for the application with  id: "
+                    + applicationId;
+            throw handleIdentityApplicationManagementException(e, msg);
+        }
+    }
+
+    public List<AuthorizedAPIResponse> getAuthorizedAPIs(String applicationId) {
+
+        try {
+            List<AuthorizedAPIResponse> authorizedAPIResponses = new ArrayList<>();
+            List<AuthorizedAPI> authorizedAPIS = getAuthorizedAPIManagementService().getAuthorizedAPIs(applicationId,
+                    PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain());
+            for (AuthorizedAPI authorizedAPI : authorizedAPIS) {
+                authorizedAPIResponses.add(new AuthorizedAPIResponse()
+                        .id(authorizedAPI.getAPIId())
+                        .identifier(authorizedAPI.getAPIIdentifier())
+                        .displayName(authorizedAPI.getAPIName())
+                        .policyId(authorizedAPI.getPolicyId())
+                        .authorizedScopes(createAuthorizedScope(authorizedAPI.getScopes())));
+            }
+            return authorizedAPIResponses;
+        } catch (IdentityApplicationManagementException e) {
+            String msg = "Error retrieving authorized APIs of application with id: " + applicationId;
+            throw handleIdentityApplicationManagementException(e, msg);
+        }
+    }
+
+    private String validatePolicy(String policyId) {
+
+        if (policyId == null || policyId.isEmpty()) {
+            // No input provided, use the default policy identifier.
+            return ApplicationManagementConstants.RBAC;
+        } else if (policyId.equalsIgnoreCase(ApplicationManagementConstants.RBAC)
```

Comment:
Always use
<constant>.equalsIgnoreCase(<variable>) pattern to avoid possible NPE.
This place we have checked policyId being null in previous step. but use this as the common practice